src/ApiAuthorizationMethod.js

```diff
@@ -125,6 +138,27 @@ export class ApiAuthorizationMethod extends AmfHelperMixin(
     });
   }
 
+  _handleCredentialsChanged(e) {
+    const { id, secret } = e.detail;
+    const headers = this[headersParam] || [];
+    const queryParameters = this[queryParametersParam] || [];
+    [headers, queryParameters].forEach(params => {
```

Comment:
Would it be better to do something like `[...headers, ...queryParameters]` and then have only a single loop?